Extensions/Video/videoruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * The PIXI.js renderer for the VideoRuntimeObject.
+ *
+ * @class VideoRuntimeObjectPixiRenderer
+ * @constructor
+ * @param {gdjs.VideoRuntimeObject} runtimeObject The object to render
+ * @param {gdjs.RuntimeScene} runtimeScene The gdjs.RuntimeScene in which the object is
+ */
+gdjs.VideoRuntimeObjectPixiRenderer = function(runtimeObject, runtimeScene) {
+  this._object = runtimeObject; // Keep a reference to the object to read from it later.
+
+  // Here we're going to create a video text as an example.
+  if (this._pixiObject === undefined) {
+    this._textureVideo = new PIXI.Texture.fromVideo(
+      "C:/Users/RTX-Bouh/Desktop/test_video_GD.mp4"
+    );
+
+    //Setup the PIXI object:
+    this._pixiObject = new PIXI.Sprite(this._textureVideo);
+
+    this._pixiObject._texture.baseTexture.source.pause();
+  }
+
+  //this._pixiObject.anchor.x = 0.5;
+  //this._pixiObject.anchor.y = 0.5;
+  runtimeScene
+    .getLayer("")
+    .getRenderer()
+    .addRendererObject(this._pixiObject, runtimeObject.getZOrder());
+
+  //this.updatePosition();
+};
+
+gdjs.VideoRuntimeObjectRenderer = gdjs.VideoRuntimeObjectPixiRenderer; //Register the class to let the engine use it.
+
+gdjs.VideoRuntimeObjectPixiRenderer.prototype.getRendererObject = function() {
+  // Mandatory, return the internal PIXI object used for your object:
+  return this._pixiObject;
+};
+
+gdjs.VideoRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function() {
+  this.updatePosition();
+};
+
+gdjs.VideoRuntimeObjectPixiRenderer.prototype.updatePosition = function() {
+  this._pixiObject.position.x = this._object.x;
+  this._pixiObject.position.y = this._object.y;
+};
+
+gdjs.VideoRuntimeObjectPixiRenderer.prototype.updateAngle = function() {
+  this._pixiObject.rotation = gdjs.toRad(this._object.angle);
+};
+
+gdjs.VideoRuntimeObjectPixiRenderer.prototype.updateOpacity = function(number) {
```

Comment:
`updateXXX` means "read from this._object and update XXX"
`setXXX` means "I give you some arguments, and you update XXX with it".

Here you're giving an argument, but the method is called updateOpacity.
So either you rename it to `setOpacity`, or read opacity from this._object for consistency :)

To stay consistent with the other methods, maybe keep "updateOpacity" but read opacity from object.

Reply:
You'll also have to divide by 255 then:
`this._pixiObject.alpha = this._object._opacity / 255`

As stated in another comment, it's usually highly discouraged to access to a property with an underscore. BUT renderers are a special case and can "read inside" their object.